src/date-adapters/luxon/index.ts

```diff
@@ -0,0 +1,187 @@
+import { DateTime } from 'luxon';
+import { DateAdapter } from '../date-adapter';
+
+export function adapterFactory(luxon): DateAdapter {
+  /* istanbul ignore next */
+  function coerceDateTime(date: Date | number): DateTime {
+    return typeof date === 'number'
+      ? luxon.DateTime.fromMillis(date)
+      : luxon.DateTime.fromJSDate(date);
+  }
+
+  function addDays(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ days: amount }).toJSDate();
+  }
+
+  function addHours(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ hours: amount }).toJSDate();
+  }
+
+  function addMinutes(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ minutes: amount }).toJSDate();
+  }
+
+  function addSeconds(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ seconds: amount }).toJSDate();
+  }
+
+  function differenceInDays(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
```

Comment:
How come we need to use `Math.trunc` here?

Reply:
I used `Math.trunc` to replicate the default behavior of date-fns and moments diff methods. Both seem to use trunc as default rounding method. Ref: https://date-fns.org/v4.1.0/docs/differenceInSeconds# and https://momentjs.com/docs/#/displaying/difference/.

> By default, moment#diff will truncate the result to zero decimal places, returning an integer.